### DISCUSSION TITLE
Can't reach database server at `aws-0-ap-southeast-1.pooler.supabase.com:6543

### DISCUSSION THREAD

**Discussion by loingtan** [prisma/prisma#23356] [Q&A]
I had tried 2 ways to make the connection:
first one conformed to the documents, it also resulted in error like that 
the second:
```
DATABASE_URL="postgres://postgres.cekcaxveddzfhnmtamgx:[YOUR PASSWORD]@aws-0-ap-southeast-1.pooler.supabase.com:6543/postgres?pgbouncer=true&connection_limit=1"
DIRECT_URL="postgres://postgres.cekcaxveddzfhnmtamgx:[YOUR PASSWORD]@aws-0-ap-southeast-1.pooler.supabase.com:5432/postgres"
```
file schema.js

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_URL")
}
```

**Comment by naeem-qv** [comment]
That is something not specific with supabase, I'm facing the same issue with prisma. I tried Neon, Supabase, vercel postgres and local postgreSQL, And found out it happens least with vercel postgres and never happens with local postgreSQL. I'm sure there's something wrong with prisma here

**Reply by janpio** [reply]
Isn't the more logical conclusion that something is wrong with the hosted database if other databases that use exctly the same code work fine?

To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by ludralph** [comment]
Hi @maikusobu 👋 

Thank you for raising this question.

Can you make sure your password doesn't have any characters in it like a `@` for instance? Also try connecting with a Postgres GUI client from your computer, something like TablePlus or BeeKeeper Studio. You may also need to [percentage encode](https://www.prisma.io/docs/orm/reference/connection-urls#special-characters) special characters in your password.

**Reply by loingtan** [reply]
I already tried it on pgadmin 4, it worked fine, my password didn't contain @ but it contained "&", "." and "_"

**Reply by janpio** [reply]
PgAdmin is not Prisma, so you will have to possible percentage encode these characters as @ludralph shared.

**Reply by loingtan** [reply]
@janpio   it still not be fixed

**Reply by ludralph** [reply]
Hi @maikusobu, You would need to percent encode the `&` character in your password. You can take a look at characters that need to be encoded [here](https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding).

**Reply by loingtan** [reply]
@ludralph , i had changed, but the issue still persists. I think there's something wrong with prisma, because when I change the network, the prisma works fine but after sometimes it becomes wrong again

**Reply by naeem-qv** [reply]
> @ludralph , i had changed, but the issue still persists. I think there's something wrong with prisma, because when I change the network, the prisma works fine but after sometimes it becomes wrong again

Exactly this happens. 👆 @ludralph

**Reply by janpio** [reply]
That sounds more like a problem connecting to Supabase from the first network then, and not a problem caused by Prisma.

When you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by naeem-qv** [comment]
> Hi @maikusobu 👋 
> 
> Thank you for raising this question.
> 
> Can you make sure your password doesn't have any characters in it like a `@` for instance? Also try connecting with a Postgres GUI client from your computer, something like TablePlus or BeeKeeper Studio. You may also need to [percentage encode](https://www.prisma.io/docs/orm/reference/connection-urls#special-characters) special characters in your password.

Hello @ludralph 👋, I wanted to share this from a long time about this strange behavior from Prisma where a message shows in the terminal exactly same as "Can't reach database server at ...."
I'm 101% sure everytime that my connection string is correct and working. So running something like Prisma generate or Prisma studio in terminal solves the problem. But the problem is we can do this in the dev environment but not in the prod. Please suggest something I've tried many services like neon,vercel postgres, supabase, railway etc... but found this issue in everyone of them with prisma.

**Reply by ludralph** [reply]
Hi @naeem-qv, what version of Prisma are you using and do you have a reproduction that you can share?

**Reply by naeem-qv** [reply]
> Hi @naeem-qv, what version of Prisma are you using and do you have a reproduction that you can share?

prisma v5.10.2, as it happens very rarely I can't give a solid reproduction for it. But one thing I'm sure that it happens with all serverless progress connection strings. I've never encountered it with localhost:5432

**Reply by janpio** [reply]
That would indicate that this is a problem with the database provider, and not Prisma.

When you have this problem the next time, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by irfnrdh** [comment]
any update?

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by naeem-qv** [comment]
> any update?

Sadly No😞

**Reply by Capovilaa** [reply]
It was fixed? I'm facing this same issue right know :(

**Reply by Capovilaa** [reply]
Are you using a company's network?

**Reply by naeem-qv** [reply]
> It was fixed? I'm facing this same issue right know :(

No it's not 😢

**Reply by Capovilaa** [reply]
I just fixed this issue changing my network connection, I was in my company one and switched for my personal

**Reply by naeem-qv** [reply]
> I just fixed this issue changing my network connection, I was in my company one and switched for my personal

Which db though?? 🤔

**Reply by janpio** [reply]
We are not aware of a problem.

To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by fabianamino** [comment]
⨯ PrismaClientInitializationError: 
Invalid `prisma.user.findUnique()` invocation:


Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com`:`6543`
I have tried everything on the internet, but to no avail

**Reply by naeem-qv** [reply]
>  ⨯ PrismaClientInitializationError: 
> Invalid `prisma.user.findUnique()` invocation:
> 
> 
> Can't reach database server at `aws-0-eu-central-1.pooler.supabase.com`:`6543`
> I have tried everything on the internet, but to no avail

Which db are you using??

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by robertoyamanaka** [comment]
Hello, I was facing the same issue and solved it encoding some values in my db password.

In my case, the DB password included **$** and **&** characters. These characters were causing connection issues due to their special meanings in URLs and environment variables. To resolve this, I URL-encoded these characters in my .env file. Specifically, I replaced $ with %24 and & with %26.


Hope this helps.

**Reply by loingtan** [reply]
mine still the same

**Reply by HarrisonRogers** [reply]
This helped a lot, thank you very much !

**Comment by Tarkbyte** [comment]
Hello, I am facing the same problem without being able to solve it, my password doesn't contain any special characters

**Reply by naeem-qv** [reply]
> Hello, I am facing the same problem without being able to solve it, my password doesn't contain any special characters

Sadly me too, my password is a simple "abcd" string but still I'm facing the issue 😔

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by sutek99** [comment]
Having this exact same issue (different AWS region).  If I remove the "&pgbouncer=true" I can actually connect to the DB until I re-start the service (all happening on local). After I restart the service I start to get the errors regarding prepared statements (which makes sense).

When I add the "&pgbouncer=true" to the connection string I get the "PrismaClientInitializationError: Can't reach database server at ____" error.

**Reply by ludralph** [reply]
Hi @sutek99 

For pgbouncer version 1.22.0 and above, can you please take a look at this recommended setup as stated in this [issue](https://github.com/prisma/prisma/issues/21531#issuecomment-1891628066)

**Comment by batmanSix** [comment]
prisma version 5.12.1 Still appear Can't reach database server at `aws-0-ap-southeast-1.pooler.supabase.com:6543

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by huseyinium** [comment]
Check out these sources: 
[https://vercel.com/guides/nextjs-prisma-postgres](https://vercel.com/guides/nextjs-prisma-postgres)
[https://supabase.com/partners/integrations/prisma](https://supabase.com/partners/integrations/prisma)

**Comment by AdarshJaiswal031** [comment]
Add the end of Database Url
``` ?pgbouncer=true ```

Update the prisma to latest. My version
``` prisma@5.13.0 ```

**Reply by janpio** [reply]
Correct, as documented on https://supabase.com/partners/integrations/prisma

**Comment by AlamMena** [comment]
I encountered a similar issue with Supabase and Prisma. I resolved it by disabling connection pooling and copying the connection string.

<img width="943" alt="Screenshot 2024-05-04 at 3 49 44 PM" src="https://github.com/prisma/prisma/assets/89660628/0c86ed62-97dd-4c6f-a4aa-6a8d8246e4dc">
pooling option on supabase

**Reply by rahul-singhania223** [reply]
Thank you,

It worked for me

**Reply by janpio** [reply]
That should not be necessary. 

If you have this problem, please open a new discussion and shared the cencored (password removed) connection strings you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by Anonymous-hss** [comment]
facing the same issue changed my password even tried changin account didn't work !!

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by JovanJevtic** [comment]
+1

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by Anonymous-hss** [comment]
Hey everyone, I kind off found the solution for those of you who where using superbase db with prisma I would suggest not to use Superbase db, Just use Neon instead. Neon gave me Database_url and everything worked fine.

**Reply by janpio** [reply]
That should not be necessary, but of course, could indicate that Supabase has problems with their connection pooler sometimes.

To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by JovanJevtic** [comment]
So, I have refreshed my supabase project and following the [Prisma docs](https://www.prisma.io/docs/orm/overview/databases/supabase) to the DATABASE_URL I added &pgbouncer=true. Also, I have updated prisma to 5.13.0. Now it's working

**Comment by lahammam** [comment]
the same error, it work in fibre optique but when I switch to another network, I get error 
Please make sure your database server is running at `aws-0-eu-central-1.pooler.supabase.com`:`5432`.

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by arewageek** [comment]
I think this has something to do with your location. I got same error while connected to a vpn but after disconnecting my queries started working all fine.

Try it out let's see if it works for you too.

**Reply by janpio** [reply]
That could indicate a problem on Supabase's network for example.

To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by DYAKOOO** [comment]
I'm having the same problem and not using any VPN. I also have added `?pgbouncer=true` but still having the same issue. Can anybody help?

**Reply by Alqorni30** [reply]
npm install -g prisma
cd path/to/your/project
npm install @prisma/client
npm install prisma --save-dev
npx prisma migrate dev

update your prisma

**Reply by yvmiko** [reply]
solved the issue by creating a new project on supabase

**Reply by Simplevhick** [reply]
please i'm currently facing this same issue 
Please make sure your database server is running at `aws-0-eu-central-1.pooler.supabase.com:6543`.
how do i get out of this

**Reply by naeem-qv** [reply]
> solved the issue by creating a new project on supabase

The problem isn't with prisma it's with server less cold starts doesn't matter neon, supabase or any other service

**Reply by naeem-qv** [reply]
> please i'm currently facing this same issue 
> Please make sure your database server is running at `aws-0-eu-central-1.pooler.supabase.com:6543`.
> how do i get out of this

You could try adding an additional param in connection string connect_timeout=20. It's default value is 10. Set it to 20 so that prisma will wait for 20 seconds before throwing the error that the db isn't reachable.

**Reply by naeem-qv** [reply]
> npm install -g prisma
> cd path/to/your/project
> npm install @prisma/client
> npm install prisma --save-dev
> npx prisma migrate dev
> 
> update your prisma
> 

It'll crash the application in production

**Reply by naeem-qv** [reply]
> I'm having the same problem and not using any VPN. I also have added `?pgbouncer=true` but still having the same issue. Can anybody help? 

Have you tried `connect_timout=20+`?

**Reply by Dev10000** [reply]
Connection string port 5432 works for me:

`DATABASE_URL="postgres://[db-user].[project-ref]:[db-password]@aws-0-[aws-region].pooler.supabase.com:5432/[db-name]"`

I tried everything and for me, the only thing that worked was the connection pooler with port 5432. Port 6543 did not work for db push and migrate. I also got a "can't reach database server at" error, but that was due to special characters in the passwords. With port 6543, there were no errors, but it did not do anything.

Prisma connection guide:
https://supabase.com/partners/integrations/prisma

**Reply by Simplevhick** [reply]
It worked already..Thank every one
I simple changed the connection string port 5432 and it workded..

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by ThisAbhinav** [comment]
I have the same error "Error: P1001: Can't reach database server at `aws-0-ap-south-1.pooler.supabase.com:5432`"

I have tried everything. Please help!

**Reply by loingtan** [reply]
I think you shoud change the network in your end, that works temporarily for me before I switch to other orms 😁, I suggest you should try drizzle or go back to traditional like typeorm

**Reply by janpio** [reply]
To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. Thank you.

**Comment by janpio** [comment]
I locked this conversation, as the discussion responses were all misses any helpful information for us to be able to figure out what is going on. I left comments on the individual threads, and please aks everyone who lands here via Google or similar to follow then:

To figure this out, when you have this problem, please [open a new discussion](https://github.com/prisma/prisma/discussions/new?category=q-a) and share the censored (password removed) connection strings (but include the hostname and other details please!) you are using for `DATABASE_URL` and `DIRECT_URL` in your `.env` file, and the command and its full output you are trying to use where you get the error message. 

Thank you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
